src/games/rcll/production.clp

```diff
@@ -485,11 +495,26 @@
 (defrule production-mps-broken
 	"The MPS is BROKEN. Inform the referee and reset the machine to stop the conveyor belt."
   (gamestate (state RUNNING) (phase PRODUCTION) (game-time ?gt))
-  ?m <- (machine (name ?n) (state BROKEN) (team ?team) (broken-reason ?reason) (broken-since 0.0))
+  ?m <- (machine (name ?n) (state BROKEN) (team ?team) (broken-reason ?reason) (broken-since 0.0)
+                 (bases-added ?ba) (bases-used ?bu) (cs-retrieved ?cap-loaded))
   =>
   (printout t "Machine " ?n " broken: " ?reason crlf)
   (assert (attention-message (team ?team) (text ?reason)))
   (modify ?m (broken-since ?gt))
+  ; Revoke points awarded for unused bases at RS slide
+  (bind ?unused-bases (- ?ba ?bu))
+  (if (> ?unused-bases 0) then
+   (bind ?deduct (* -1 (min ?unused-bases ?*LOADED-WITH-MAX*)))
```

Comment:
Why do you need `(min ?unused-bases ?*LOADED-WITH-MAX*)` instead of just `?unused-bases`?

Reply:
That's right, we no longer need this. I applied the changes, thanks for pointing this out!

(Reason:  Previously, in the old code, it used to be the case that the
    bases-added slot was always updated in the machine fact by the
    SLIDE COUNTER, keeping track of every base add operation, even
    if the recently added base was above the capacity. The machine
    would later break but the added-bases slot difference to the
    used-bases will end up more the max capacity. Hence, it was needed
    to make sure we do not deduct more points than we need.

    This is no longer the case in the refactored code. The bases-added is
    updated on an add event in the machine fact only if,the capacity isn't
    violated. This makes sure that only usable bases are tracked.)